Feature/translation script

yjmrobert opened the pull request:
Wrote a console app to parse all the `.razor` and `.cs` files for any strings inside `Localizer[`".+ ?"` and then cross references all of those into the i18n json files.

In order to simplify everything, the i18n json translations were consolidated into a single file for each locale (coincidentally removing a few dozen duplicate keys).

Proposed workflow for developers would now be:
- continue writing all translatable text with the `Localizer`
- executing the `Datahub.MissingTranslations` console app
- completing the french translations for all the generated text/labels
- pasting the `en` json output into the `en` i18n json
- pasting the translated `fr` json output into the `fr` i18n json
- sorting both the i18n json files

yjmrobert committed:
🔧 Consolidated i18n files

Consolidated all the translation files into a single json for each locale. Removed all duplicate keys.

yjmrobert committed:
🔧 Removed legacy nested translations

yjmrobert committed:
🔧 Sorted the translations

yjmrobert committed:
🔧 Add French translation handling in missing translations check

The script for checking and generating missing translations has been extended to also handle French translations. Specifically, the modifications not only check for missing English translations, but also for French ones. Additionally, the results for both languages are now outputted to individual files to maintain them separately.

yjmrobert committed:
📜 Added all missing translations

ErikApption commented on Portal/src/Datahub.Portal/Datahub.Portal.csproj:
```diff
@@ -68,4 +68,9 @@
 		<ProjectReference Include="..\modules\Datahub.PIP\Datahub.PIP.csproj" />
 	</ItemGroup>
+	<ItemGroup>
+	  <_ContentIncludedByDefault Remove="i18n\nrcan\initiatives.fr.json" />
```
any reason we don't remove the file instead?

yjmrobert replied:
removed

yjmrobert committed:
📜 Remove unnecessary content inclusion in Datahub.Portal.csproj

The removed lines in the .csproj file were including unnecessary Json files from the i18n\nrcan directory. This likely led to an unwanted increase in the compiled project's size and potential confusion in the code base.

yjmrobert committed:
Change output filename for English translations

The output filename for English translations in the Datahub.MissingTranslations program has been modified. Previous code was overwriting the file with English translations due to the same filename. The change ensures proper segregation and avoids orverridding.

yjmrobert committed:
🐛 Remove profile-based configuration and localization

The commit removes the profile-based application configuration and localization source setup from the startup method. This change reduces redundancy and simplifies code by eliminating the specific reference to the DataHub profile. The removed reference is associated with the environment's hosting profile, which is no longer being utilized.

yjmrobert committed:
🐛 added missing comma

Sean-Stilwell reviewed (approved):
"Mes espaces de travail" wraps on the French sidebar now and that truly ruins my day.

Otherwise 11/10 awesome work

akhavich commented on Portal/src/Datahub.Portal/i18n/localization.json:
```diff
@@ -1,1272 +1,1540 @@
+  "CIOSB": "CIOSB",
+  "CMS - Others": "CMS - Others",
+  "CMSS": "CMSS",
+  "COMMON.No": "COMMON.No",
```
Hierarchical translations like this seem to have all been collapsed and replaced with their keys instead of the values they had. Not sure if any of the existing ones are still in use anywhere in the app, but for the Open Data stuff I'm working on I did use the hierarchy. E.g. TbsOpenPublishing.Status.[each status].Title and .Description. I can manually update those ones in the json file for the work I'm doing now, but if we have e.g. COMMON.Yes or ERROR-PAGE.whatever, we would need to dig through the file history to restore the translated values.

yjmrobert replied:
Yes, please switch to not using the hierarchy.

I believe most of the hierarchy formatted keys were used on legacy components, however any remaining will be obvious enough to fix quickly.
